meson.build

```diff
@@ -1570,6 +1570,18 @@ if features['cocoa'] and features['swift']
                            'video/out/mac/window.swift')
 endif
 
+macos_11_features = get_option('macos-11-features').require(
+    macos_sdk_version.version_compare('>= 11'),
+    error_message: 'A macos sdk version >= 11 could not be found!',
+)
+features += {'macos-11-features': macos_11_features.allowed()}
+
+macos_12_features = get_option('macos-12-features').require(
+    macos_sdk_version.version_compare('>= 12'),
+    error_message: 'A macos sdk version >= 12 could not be found!',
+)
+features += {'macos-12-features': macos_12_features.allowed()}
```

Comment:
Also is this a best way to check SDK version, instead of actually definitions during build?

`__MAC_OS_X_VERSION_MAX_ALLOWED` or something like that.

Reply:
sure it does support building on lower SDKs. not every runtime check needs a build time check. the ones you mentioned are just a workaround for older version of macOS, the properties and variables used are available on all supported macOS versions.

to my knowledge this is the best way to check it. since SDK version is not the same as max allowed version. though i think re-evaluating this is something out of scope of this PR.